src/ThreadPool.cc

```diff
@@ -0,0 +1,327 @@
+/**
+ * \file    ThreadPool.cc
+ * \authors \ref contributors
+ * \date    Created on November 6, 2018
+ * \brief   Implementation of the ThreadPool class.
+ *
+ * \copyright Copyright 2018-2024 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <algorithm>
+#include <iostream>
+
+#include <gul14/cat.h>
+#include <gul14/ThreadPool.h>
+#include <gul14/time_util.h>
+
+#if defined(__APPLE__) || defined(__GNUC__)
```

Comment:
Maybe define a helper macro, that can be used below (line 224)

Reply:
That is a possibility. On the other hand, even Windows has a `signal.h`, so maybe we can just include it without any guard. I'll try.

Reply:
But I believe Windows' `std::thread` is not based on `pthread`? :woman_shrugging: 

I just meant the macro to be always consistent, THREAD_POOL_USE_SIGNAL, and then do what is needed for that and not checking the rules for the decision to use or not use signal-suppression in multiple places.

Reply:
The `#if defined(__APPLE__) || defined(__GNUC__)` now only occurs in a single place. Should we still define a THREAD_POOL_USE_SIGNAL macro for it? That is one more macro to track...

Reply:
No - the comment was just for enforcing consistency. Do a test like `if A and B` in more than one (two) places requires us to keep track of that places to possibly change them all in unison. That could be made 'grepable' and enforced wit the additional macro.
If there is only one place my comment is wrong. Maybe you changed it meanwhiles. Ah it says outdated.
Closing this. :+1: